src/neptune_mlflow_exporter/impl/components/exporter.py

```diff
@@ -0,0 +1,100 @@
+#
+# Copyright (c) 2023, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+__all__ = ["Exporter"]
+
+from datetime import datetime
+from typing import Optional
+
+import mlflow
+from mlflow.entities import Experiment
+from mlflow.entities import Run as MlflowRun
+
+try:
+    from neptune import Run as NeptuneRun
+except ImportError:
+    from neptune.new.metadata_containers import Run as NeptuneRun
+
+from neptune_mlflow_exporter.impl.strategies import (
+    DirectoryUploadStrategy,
+    FileUploadStrategy,
+)
+
+
+class Exporter:
+    def __init__(self, client: mlflow.tracking.MlflowClient):
+        self.mlflow_client = client
+
+        self._cached_experiment: Optional[Experiment] = None
+
+    def export_experiment_metadata(self, neptune_run: NeptuneRun, mlflow_run: MlflowRun) -> None:
+        if self._cached_experiment and self._cached_experiment.experiment_id == mlflow_run.info.experiment_id:
```

Comment:
Are we processing runs based on a sorted experiments list? As I don't think this will work if not.